src/core/tests/TestAptitudePackageManager.py

```diff
@@ -111,5 +119,121 @@ def test_install_package_only_upgrades(self):
         # test for unsuccessfully installing a package
         self.assertEquals(package_manager.install_update_and_dependencies('iucode-tool', '1.5.1-1ubuntu0.1', simulate=True), Constants.PENDING)
 
+    def test_get_current_auto_os_update_settings_success(self):
+        self.runtime.set_legacy_test_type('HappyPath')
+        package_manager = self.container.get('package_manager')
+        current_auto_os_update_settings = package_manager.get_current_auto_os_update_settings()
+        self.assertTrue(current_auto_os_update_settings is not None)
+
+    def test_get_current_auto_os_update_settings_fail(self):
+        package_manager = self.container.get('package_manager')
+
+        self.runtime.set_legacy_test_type('SadPath')
+        current_auto_os_update_settings = package_manager.get_current_auto_os_update_settings()
+        self.assertTrue(current_auto_os_update_settings is "")
+
+        self.runtime.set_legacy_test_type('ExceptionPath')
+        current_auto_os_update_settings = package_manager.get_current_auto_os_update_settings()
+        self.assertTrue(current_auto_os_update_settings is None)
+
+    def test_disable_auto_os_update_success(self):
+        self.runtime.set_legacy_test_type('HappyPath')
+        package_manager = self.container.get('package_manager')
+        package_manager.disable_auto_os_update()
+        self.assertTrue(package_manager.default_auto_os_update_log_exists())
+        default_auto_os_update = self.runtime.env_layer.file_system.read_with_retry(package_manager.default_auto_os_update_log_path)
+        self.assertTrue(default_auto_os_update is not None)
+        self.assertTrue('APT::Periodic::Update-Package-Lists "1"' in default_auto_os_update)
+        self.assertTrue('APT::Periodic::Unattended-Upgrade "1"' in default_auto_os_update)
+
+    def test_disable_auto_os_update_failure(self):
+        # disable with non existing log file
+        self.runtime.set_legacy_test_type('SadPath')
```

Comment:
I don't see the specified path used inside legacyenvlayerextensions... reminder to discuss. (Etc. in other tests).

Reply:
TODO: Review again based on disucssion.

Reply:
Not using this path in per the current implementation